tests/src/isometric/unary/math_helpers.cpp

```diff
@@ -265,3 +319,490 @@ TEST_F(MathTest, RoundByColumn) {
+    tatami::DelayedCoshHelper op;
+    auto dense_mod = tatami::make_DelayedUnaryIsometricOp(dense, op);
+    auto sparse_mod = tatami::make_DelayedUnaryIsometricOp(sparse, op);
+
+    EXPECT_FALSE(dense_mod->sparse());
+    EXPECT_FALSE(sparse_mod->sparse());
+    EXPECT_EQ(dense->nrow(), dense_mod->nrow());
+    EXPECT_EQ(dense->ncol(), dense_mod->ncol());
+
+    auto refvec = simulated;
+    for (auto& r : refvec) {
+        r = std::cosh(r);
+    }
+    tatami::DenseRowMatrix<double> ref(nrow, ncol, std::move(refvec));
+
+    // Again, doing some light tests.
+    bool FORWARD = true;
+    int JUMP = 1;
+
+    tatami_test::test_simple_column_access(dense_mod.get(), &ref, FORWARD, JUMP);
+    tatami_test::test_simple_column_access(sparse_mod.get(), &ref, FORWARD, JUMP);
+
+    tatami_test::test_simple_row_access(dense_mod.get(), &ref, FORWARD, JUMP);
+    tatami_test::test_simple_row_access(sparse_mod.get(), &ref, FORWARD, JUMP);
+}
+
+TEST_F(MathTest, TanhByColumn) {
+    tatami::DelayedTanhHelper op;
+    auto dense_mod = tatami::make_DelayedUnaryIsometricOp(dense, op);
+    auto sparse_mod = tatami::make_DelayedUnaryIsometricOp(sparse, op);
+
+    EXPECT_FALSE(dense_mod->sparse());
+    EXPECT_TRUE(sparse_mod->sparse());
+    EXPECT_EQ(dense->nrow(), dense_mod->nrow());
+    EXPECT_EQ(dense->ncol(), dense_mod->ncol());
+
+    auto refvec = simulated;
+    for (auto& r : refvec) {
+        r = std::tanh(r);
+    }
+    tatami::DenseRowMatrix<double> ref(nrow, ncol, std::move(refvec));
+
+    // Again, doing some light tests.
+    bool FORWARD = true;
+    int JUMP = 1;
+
+    tatami_test::test_simple_column_access(dense_mod.get(), &ref, FORWARD, JUMP);
+    tatami_test::test_simple_column_access(sparse_mod.get(), &ref, FORWARD, JUMP);
+
+    tatami_test::test_simple_row_access(dense_mod.get(), &ref, FORWARD, JUMP);
+    tatami_test::test_simple_row_access(sparse_mod.get(), &ref, FORWARD, JUMP);
+}
+
+TEST_F(MathTest, AsinhByColumn) {
+    tatami::DelayedAsinhHelper op;
+    auto dense_mod = tatami::make_DelayedUnaryIsometricOp(dense, op);
+    auto sparse_mod = tatami::make_DelayedUnaryIsometricOp(sparse, op);
+
+    EXPECT_FALSE(dense_mod->sparse());
+    EXPECT_TRUE(sparse_mod->sparse());
+    EXPECT_EQ(dense->nrow(), dense_mod->nrow());
+    EXPECT_EQ(dense->ncol(), dense_mod->ncol());
+
+    auto refvec = simulated;
+    for (auto& r : refvec) {
+        r = std::asinh(r);
+    }
+    tatami::DenseRowMatrix<double> ref(nrow, ncol, std::move(refvec));
+
+    // Again, doing some light tests.
+    bool FORWARD = true;
+    int JUMP = 1;
+
+    tatami_test::test_simple_column_access(dense_mod.get(), &ref, FORWARD, JUMP);
+    tatami_test::test_simple_column_access(sparse_mod.get(), &ref, FORWARD, JUMP);
+
+    tatami_test::test_simple_row_access(dense_mod.get(), &ref, FORWARD, JUMP);
+    tatami_test::test_simple_row_access(sparse_mod.get(), &ref, FORWARD, JUMP);
+}
+
+TEST_F(MathTest, AcoshByColumn) {
+    double CONSTANT = 11;
```

Comment:
Explain why we need a `CONSTANT` here, presumably to bring the values into range.

Reply:
The domain of `acosh` is `x >= 1`. Since the simulated values are in `[-10, 10]`, `11` is added to the test values

Reply:
As in, add this in a comment.

Reply:
Same for that other use of `CONSTANT`.

Reply:
Added comments for constants in tests